### ISSUE TITLE
Breaking change between 9.1.3 and 9.1.4 with Docusaurus

### ISSUE THREAD

**Issue by EugeneLeclerc** [mermaid-js/mermaid#3428]
Hi! I would like to first thank you for Mermaid, I use it a lot and I've been happy with it for a long time.

**Describe the bug**

I'm using Mermaid with docusaurus (2.1.0) and mdx-mermaid (1.3.2), both latest versions. Between mermaid-js 9.1.3 and 9.1.4, something changed in mermaid-js that broke my Git Graphs. The generated SVG is simply empty, with no error displayed anywhere. I'm conscious the problem may be somewhere in mdx-mermaid, but I did not expect a simple minor version change to change behaviour.

**To Reproduce**

1. Create a simple Docusaurus project and use [mdx-mermaid](https://github.com/sjwall/mdx-mermaid/tree/v1.x.x) with it.
2. Write a simple Git Graph in one of your Docusaurus page like [the one from your doc](https://mermaid-js.github.io/mermaid/#/gitgraph)
![image](https://user-images.githubusercontent.com/44815069/188883383-58196b58-f00a-4c39-9cf4-8621403b6a0d.png)
3. On 9.1.3, the graph works well:
![image](https://user-images.githubusercontent.com/44815069/188883767-6c53214a-bb45-4162-8a81-23056a42fe9f.png)
4. On 9.1.4, it does not:
![image](https://user-images.githubusercontent.com/44815069/188883548-16015fbb-d12b-4a7b-8295-6ebb7657b4e6.png)
On 9.1.4, the generated SVG is empty:
![image](https://user-images.githubusercontent.com/44815069/188884788-54afcf46-1a61-4486-84fd-b61f4f9ee215.png)

**Expected behavior**
I expect the Git Graph to show properly. If there is a problem, I would expect an error showing somewhere. Even if the problem is with mdx-mermaid or docusaurus, I would not expect a patch version change in mermaid-js to break anything.

**Platform:**
Probably not relevant, but:
- OS: Windows
- Browser: Both Chrome 104 and Firefox 104

Once again, thanks for your work on Mermaid! I do not have time right now to investigate further, but I will try to find the problematic commit later.

**Comment by aloisklink** [comment]
Does this also affect the latest version of `mermaid`, v9.1.5 or v9.1.6?

I also had issues upgrading from v9.1.3 to v9.1.4, however this was fixed in v9.1.5.

I'm the creator of [remark-mermaid-dataurl](https://github.com/aloisklink/remark-mermaid-dataurl), which is a similar plugin for adding mermaid to docusaurus (although for docusaurus, `mdx-mermaid` is usually better than my plugin, since `mdx-mermaid` does client-side rendering).

**Comment by EugeneLeclerc** [comment]
Yes, this behavior is also present in 9.1.5 and 9.1.6. I initially found it by upgrading from 9.1.1 to 9.1.6 and worked backward through the version history to find the breaking point (9.1.3 -> 9.1.4).

**Comment by knsv** [comment]
OK. First of all we need to reproduce this issue. Any hints on how to go about that would be appreciated. 😄 

We then fix the issue in the 9.1.7 release branch,

**Comment by EugeneLeclerc** [comment]
I will try to create you a sample project with the bug when I get the time in the next few days. Thanks for the answer!

**Comment by knsv** [comment]
@EugeneLeclerc I managed to reproduce the issue. It has to do with work decoupling the diagrams from the core of Mermaid and with how mdx-mermaid use Mermaid. The decoupled diagrams are loaded when `initialize` is called. In the plugin I believe that  `initialize` is only called when config is supplied.

I could make a gitgraph appear when suppling a theme for instance. When I did not supply the config, initialise was never called. I have updated Mermaid so that the loosely coupled diagrams are called even without the `initialize` call. This fix will be released in 9.1.7
